src/main/java/com/amplitude/Amplitude.java

```diff
@@ -51,16 +56,42 @@ public static Amplitude getInstance(String instanceName) {
   }
 
   /**
-   * Set the API key for this instance of Amplitude. API key is necessary to authorize
-   * and route events to the current Amplitude project.
+   * Set the API key for this instance of Amplitude. API key is necessary to authorize and route
+   * events to the current Amplitude project.
+   *
    * @param key the API key from Amplitude website
    */
   public void init(String key) {
     apiKey = key;
+    updateHttpCall(HttpCallMode.REGULAR_HTTPCALL);
+  }
+
+  /**
+   * Set the Event Upload Mode. If isBatchMode is true, the events will log through the Amplitude
+   * HTTP V2 Batch API.
+   *
+   * @param isBatchMode if using batch upload or not;
+   */
+  public void useBatchMode(Boolean isBatchMode) {
+    updateHttpCall(isBatchMode ? HttpCallMode.BATCH_HTTPCALL : HttpCallMode.REGULAR_HTTPCALL);
+  }
+
+  private void updateHttpCall(HttpCallMode updatedHttpCallMode) {
+    if (httpCallMode == null) {
```

Comment:
oh, originally I was thinking about httpCall null check, as you define httpCallMode in constructor. If you do it this way, it is ok too, just you can skip the null check, or combine them together.

`if (httpCallMode == null || htppCallMode != updatedHttpCallMode)`